### PR TITLE
Implements 18592 QoS Guarantee for pull-kubernetes-files-remake

### PR DESCRIPTION
**Pull request by RobertKielty**
Adds matching limits and requests for cpu and memory : 1500m cpu, 7Gi mem

Part of #18530
cc @kubernetes/ci-signal

### CODE REVIEW

File: config/jobs/kubernetes/sig-testing/kubernetes-kind-presubmits.yaml [hunk 1]
```diff
@@ -283,13 +283,12 @@ presubmits:
         securityContext:
           privileged: true
         resources:
+          limits:
+            cpu: 4
+            memory: 9Gi
```
Comment (spiffxp) [inline config/jobs/kubernetes/sig-testing/kubernetes-kind-presubmits.yaml:288]: This should be a different PR

Comment (RobertKielty) [reply config/jobs/kubernetes/sig-testing/kubernetes-kind-presubmits.yaml:288]: I'll amend
